### PR TITLE
NAS-114945 / 22.02.1 / Expand warning for shell modifications

### PR DESCRIPTION
**Pull request by anodos325**
"CLI" is now ambiguous because it refers to the TrueNAS CLI (which
is a supported way of making changes).

### CODE REVIEW

File: src/freenas/root/.warning [hunk 1]
```diff
@@ -1,4 +1,8 @@
 
-Warning: settings changed through the CLI are not written to
-the configuration database and will be reset on reboot.
+Warning: the only supported method of making changes to the TrueNAS
+configuration is through the TrueNAS CLI or the TrueNAS REST and Websocket APIs.
```
Comment (themylogin) [inline src/freenas/root/.warning:3]: Can we say "TrueNAS UI, TrueNAS CLI and TrueNAS APIs"?

File: src/freenas/root/.warning [hunk 2]
```diff
@@ -1,4 +1,8 @@
...
+Manual configuration changes through the system shell will not persist after
+reboot and may not persist after service reload or restart. Altering the base
+install of TrueNAS through installing new packages is also not supported and
```
Comment (ericloewe) [inline src/freenas/root/.warning:6]: I would add updating and removing to installing. E.g.
"... through installing new packages or updating or removing existing packages ..."

**Comment by yocalebo** [comment]
retest this please

File: src/freenas/root/.warning [hunk 3]
```diff
@@ -1,8 +1,8 @@
...
-configuration is through the TrueNAS CLI or the TrueNAS REST and Websocket APIs.
-Manual configuration changes through the system shell will not persist after
-reboot and may not persist after service reload or restart. Altering the base
-install of TrueNAS through installing new packages is also not supported and
-may result in breaking the TrueNAS appliance.
+Warning: the only supported mechanisms for TrueNAS configuration changes
+are the TrueNAS WebUI, TrueNAS CLI, and TrueNAS API. Manual configuration
+changes through the system shell will not persist after reboot and may not
+persist after service reload or restart. Altering the base install of
+TrueNAS install through installing, updating, or removing packages is not
```
Comment (ericloewe) [inline src/freenas/root/.warning:6]: One "install" too many: Either "Altering the base install of TrueNAS" or "Altering the TrueNAS base install"

Comment (anodos325) [reply src/freenas/root/.warning:6]: Indeed.

**Review by yocalebo — APPROVED** [review]

File: src/freenas/root/.warning [hunk 4]
```diff
@@ -1,8 +1,6 @@
 
-Warning: the only supported mechanisms for TrueNAS configuration changes
-are the TrueNAS WebUI, TrueNAS CLI, and TrueNAS API. Manual configuration
-changes through the system shell will not persist after reboot and may not
-persist after service reload or restart. Altering the base install of
-TrueNAS install through installing, updating, or removing packages is not
-supported and may result in breaking the TrueNAS appliance.
+Warning: the supported mechanisms for making configuration changes
+are the TrueNAS WebUI, CLI, and API exclusively. ALL OTHERS ARE
+NOT SUPPORTED AND WILL RESULT IN UNEXPECTED BEHAVIOR AND MAY
```
Comment (ericloewe) [inline src/freenas/root/.warning:4]: I'm partial to "undefined" behavior instead of "unexpected".

**Comment by stephendt** [comment]
Should probably specifically mention that the usage of package managers for updates is likely to break their system, just in case someone wants to configure automatic updates via CRON or something.

**Pull request merged**